woocommerce/woocommerce-gutenberg-products-block
# More API price formatting and currency handling

This took a lot longer than I'd hoped, but this should now be in a better place.

This PR introduces a consistent response format for endpoints returning prices/totals via a new method `get_store_currency_properties` and `get_store_currency_response`. When prices are returned, currency and minor unit information is also returned. This mostly affects product and cart endpoints.

As well as this, other endpoints handling prices have been updated. Price filtering (and active price filters), and product price templates being the main ones. These now also deal with subunit/cent values, and formatting is taken care of using the `react-number-format` library where possible.

### How to test the changes in this Pull Request:

1. Check API responses for /cart/ `totals` and product `prices` show correct pricing data and return the currency.
2. Test that you can filter by price using the all products block, and that prices are shown in the correct format.
3. Check using other currencies in WooCommerce, and/or change the number of decimals in settings.

Next steps:
- After merge, update coupons PR to follow these patterns.
- Add Todos for things which could use improvement. I noticed the CURRENCY setting uses `precision` and other terminology which could better match the API. Additionally, using prefix/suffix for currency seems to be easier to handle that sprintf/`priceFormat` as we have in places.
